### PR TITLE
Wait for execution to finish before exitting

### PR DESCRIPTION
**Pull request by tanmayv25**
Most likely fixes L0_decoupled test.

### CODE REVIEW

File: src/backends/backend/triton_model_instance.cc [hunk 1]
```diff
@@ -130,10 +130,15 @@ TritonModelInstance::TritonModelInstance(
         &reporter_);
   }
 #endif  // TRITON_ENABLE_METRICS
+  ongoing_requests_count_.store(0);
 }
 
 TritonModelInstance::~TritonModelInstance()
 {
+  while (ongoing_requests_count_.load() != 0) {
+    LOG_VERBOSE(1) << "Waiting for the execution on instance to finish...";
```
Comment (CoderHam) [inline src/backends/backend/triton_model_instance.cc:139]: You should add the Instance / Model name here

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: src/backends/backend/triton_model_instance.cc [hunk 2]
```diff
@@ -580,6 +587,8 @@ TritonModelInstance::Execute(
 
     TRITONSERVER_ErrorDelete(err);
   }
+
+  ongoing_requests_count_--;
```
Comment (GuanLuo) [inline src/backends/backend/triton_model_instance.cc:591]: The decrement is done in `Execute()`, if the request is delegated to a separate thread, Does the `count` track the ongoing request properly? In this case, the `inst_exec_fn` returns "immediately", right?

**Comment by tanmayv25** [comment]
Closing this PR as there might be a better way to handle the clean up. I am pretty sure where the race condition is.

File: src/backends/backend/triton_model_instance.cc (hunk 2 above)
Comment (tanmayv25) [reply src/backends/backend/triton_model_instance.cc:591]: It doesn't return immediately. We have subtests that have WAIT time specified as large as 10 sec. That being said I was too much focused on triaging the issue that I came up with this very specific solution. I will share a proper fix soon.

**Pull request closed without merging**